### ISSUE TITLE
[Accessibility]: Global nav bar: Extensions item unexpectedly steals focus

### ISSUE THREAD

**Issue by umpox** [sourcegraph/sourcegraph#35114]
### Audit type

Keyboard navigation

### User journey audit issue

https://github.com/sourcegraph/sourcegraph/issues/34411

### Problem description

1. Navigate through the navigation bar
2. Select "Extensions"
3. See that focus has been unexpectedly removed.

### Expected behavior

Although this is expected in some cases (e.g. "Code search" naturally can focus a "search" bar) - it isn't expected for just "Extensions". We should remove autofocus from this page.

### Assigning labels

- [X] Please give this issue an estimate by applying a label like `estimate/Xd`, where X is the estimated number of days it will take to complete.
- [X] If this issue is specific to a specific Sourcegraph product, please assign the appropriate team label to this issue.
- [ ] If this issue will require input from designers in order to complete, please assign the label `needs-design`.
- [ ] If you are confident that this issue should be fixed by GitStart, please assign the label `gitstart`.

### Owner

This issue will be fixed by my team, I have assigned a relevant member to it, or I will do so in the near future.
